### PR TITLE
Handle all kinds of OBS Links in the token controller

### PR DESCRIPTION
**Pull request by hennevogel**
- Remote projects
- Project with a project-link to a Project
- Project with a project-link to a remote Project
- Multibuild flavors

### CODE REVIEW

File: src/api/app/controllers/trigger_controller.rb [hunk 1]
```diff
@@ -9,47 +9,86 @@ class TriggerController < ApplicationController
...
+
+    # We found a local package
+    @token.object_to_authorize = if @package.is_a?(Package)
+                                   # If the package is coming through a project link, we authorize the project
+                                   # See https://github.com/openSUSE/open-build-service/wiki/Links#project-links
+                                   package_from_project_link? ? @project : @package
+                                   # We did not find a local package, have to authorize the project
+                                 else
+                                   @project
+                                 end
```
Comment (hennevogel) [inline src/api/app/controllers/trigger_controller.rb:88]: This doesn't take permissions for a local multibuild package into account.

File: src/api/spec/models/token/release_spec.rb [hunk 2]
```diff
@@ -10,20 +10,20 @@
...
   let(:target_repository) { create(:repository, name: 'target_repository', project: target_project) }
 
-  subject { create(:release_token, :with_package_from_association_or_params, package: package, user: user) }
+  subject { create(:release_token, package: package, user: user) }
 
   describe '#call' do
     context 'when no release target is set' do
       it 'throws an exception' do
-        expect { subject.call({}) }.to raise_error(Token::Errors::NoReleaseTargetFound, 'Bar:Staging has no release targets that are triggered manually')
+        expect { subject.call(package: package) }.to raise_error(Token::Errors::NoReleaseTargetFound, 'Bar:Staging has no release targets that are triggered manually')
```
Comment (hennevogel) [inline src/api/spec/models/token/release_spec.rb:18]: We can also change `subject`

File: src/api/spec/controllers/trigger_controller_spec.rb [hunk 3]
```diff
@@ -201,26 +185,127 @@
     end
   end
 
+  describe '#set_project' do
+    let(:signature_header_name) { 'HTTP_X_OBS_SIGNATURE' }
```
Comment (krauselukas) [inline src/api/spec/controllers/trigger_controller_spec.rb:189]: @hennevogel the `signature_header_name` seems not be used

File: src/api/spec/controllers/trigger_controller_spec.rb [hunk 4]
```diff
@@ -201,26 +185,127 @@
...
+      params = { project: 'does-not-exist', package: package.name, format: :xml }
+      post :create, params: params
+      expect(response).to have_http_status(:not_found)
+    end
+  end
+
   describe '#set_package' do
-    context 'when entity does not exist' do
-      let(:signature_header_name) { 'HTTP_X_OBS_SIGNATURE' }
+    let(:signature_header_name) { 'HTTP_X_OBS_SIGNATURE' }
```
Comment (krauselukas) [inline src/api/spec/controllers/trigger_controller_spec.rb:210]: Same here, can't the see it being used

**Review by eduardoj — APPROVED** [review]

**Review by krauselukas — APPROVED** [review]

**Pull request merged**